waitfor.go

```diff
@@ -645,6 +665,10 @@ func (p *EventPoller) WaitForLatestUnknownEvent(ctx context.Context) (*Event, er
 			}
 
 			for _, event := range events {
+				if !eventMatchesSecondary(p.SecondaryEntityID, event) {
+					continue
+				}
+
```

Comment:
Is adding an additional API filter when `SecondaryEntityID != nil` a better and workable idea?